keep-ui/app/navbar.tsx

```diff
@@ -6,10 +6,17 @@ import { useSession } from "next-auth/react";
 import { CMDK } from "./command-menu";
 
 export default function Navbar() {
-  const { data: session } = useSession();
+  // extract status and session properties from useSession hook
+  const { status, data: session } = useSession();
+  // if session is loading, return nothing
+  if (status === "loading") return <></>;
+  if (status === "unauthenticated") return <></>;
+  if (!session) return <></>;
+
+
   return (
     <PHProvider>
-      <NavbarInner user={session?.user} />
+      <NavbarInner session={session}/>
```

Comment:
Based on this logic, `session` should always be defined. I think removing `undefined` from the `session` prop should help make that obvious in `NavbarInner`. Unless, of course, it's being used elsewhere...